--- jdic-0.9.3-src/jdic/src/share/classes/org/jdesktop/jdic/init/JdicManager.java	2007-07-13 16:16:25.000000000 +0200
+++ jdic-0.9.3-src.n/jdic/src/share/classes/org/jdesktop/jdic/init/JdicManager.java	2007-09-12 15:09:17.000000000 +0200
@@ -30,7 +30,7 @@
 
 import org.jdesktop.jdic.browser.internal.WebBrowserUtil;
 
-import com.sun.jnlp.JNLPClassLoader;
+//import com.sun.jnlp.JNLPClassLoader;
 
 /**
  * Initialization manager for JDIC to set the environment variables or
@@ -210,18 +210,18 @@
     {
         String jdicLibFolder = null;
         ClassLoader cl = this.getClass().getClassLoader();
-        if (cl instanceof JNLPClassLoader) {
+        /* if (cl instanceof JNLPClassLoader) {
             JNLPClassLoader jnlpCl = (JNLPClassLoader) cl;
             String jdicLibURL = jnlpCl.findLibrary("jdic");//get lib path by classloder
             jdicLibFolder = (new File(jdicLibURL)).getParentFile().getCanonicalPath();
             WebBrowserUtil.trace("running path " + nativeLibPath);
             isShareNativeInitialized = true;
-        } else {
+        } else { */
             // only run well for sun jre
             throw new JdicInitException(
                 "Unexpected ClassLoader for webstart, only com.sun.jnlp.JNLPClassLoader is supported.");
-        }
-        return jdicLibFolder;
+        //}
+        //return jdicLibFolder;
     }
 
     /**
